middleware.js

```diff
@@ -69,10 +74,7 @@ function middleware (api, { baseIriFromRequest, loader, store, middleware = {} }
     throw new Error('no loader or store provided')
   }
 
-  if (middleware.resource) {
-    router.use(waitFor(init, () => middleware.resource))
-  }
-  router.use(waitFor(init, () => operation(api)))
+  router.use(waitFor(init, () => operation(api, middleware.resource)))
```

Comment:
I moved the resource middleware inside the operation because the `resource` handler above may not set a single `req.hydra.resource`.